src/content-scripts/canvas.js

```diff
@@ -1,98 +1,168 @@
+    ref = new Firebase('https://dazzling-heat-2465.firebaseio.com/web/data/sites/' + tabUrl);
+    currentUser;
 
+// Message Handler
 chrome.runtime.onMessage.addListener(
   function (request, sender, sendResponse) {
-    console.log('message:', request);
+    console.log('message:', request, ' from sender: ', sender);
+
+    // Toggle User Canvas Messages
     if (request.toggle === 'off') {
-        toggleCanvasOff();
-        toggle = 'off';
+        toggleUserCanvasOff();
         sendResponse({confirm:'canvas turned off'});
     } else if (request.toggle === 'on') {
-        toggleCanvasOn();
-        toggle = 'on';
+        toggleUserCanvasOn();
+        getFirebaseAuthData();
         sendResponse({confirm:'canvas turned on'});
     } else if (request.getStatus === true) {
       sendResponse({status:toggle});
+
+    // Logout Messages
+    } else if (request.logout) {
+      removeCanvasAll();
+
+    // Show Public Canvases Messages
+    } else if (request.show === 'all') {
+      showCanvasAll === true;
+    } else if (request.show === 'none') {
+       showCanvasAll === false;
+       removePublicCanvasAll();
     }
   }
 );
 
-function toggleCanvasOn(){
-  if (toggle === 'off') {
-    $('<canvas id="graffio-canvas"></canvas>')
-      .css(overlayPage)
-      .attr('width', document.body.scrollWidth) // sets to max width
-      .attr('height', document.body.scrollHeight) // sets to max height
-      .on('mousemove', function(e){findxy('move', e)})
-      .on('mousedown', function(e){findxy('down', e)})
-      .on('mouseup', function(e){findxy('up', e)})
-      .on('mouseout', function(e){ findxy('out', e)})
-      .appendTo('body');
+// Firebase Event Listener 
+ref.on("value", function(snapshot){
+  if (showCanvasAll) {
+    console.log('value')
+    updateCanvasElements(snapshot);
+  } 
+})
 
-    canvas = document.getElementById('graffio-canvas');
-    ctx = canvas.getContext("2d");
-    console.log('canvas injected!');
+function getFirebaseAuthData(){
+  chrome.runtime.sendMessage({action: 'getToken'}, function(response) {
+    if (response.token) {
+      ref.authWithCustomToken(response.token, function(error, result) {
+        if (error) {console.log("Login Failed!", error);} 
+        else {currentUser=result.uid;}
+      });
+    } else {
+      console.log('no token found');
+    }
+  });
+};
+
+function toggleUserCanvasOn(){
+  if (toggle === 'off') {
+    appendCanvasElement(currentUser);
+    toggle = 'on';
   }
 };
 
-function toggleCanvasOff(){
+function toggleUserCanvasOff(){
   $('canvas#graffio-canvas').remove();
-  console.log('canvas removed!');
+  toggle = 'off';
+  console.log('user canvas removed!');
+};
+
+function saveUserCanvas(){
+  console.log('saving user canvas')
+  var data = canvas.toDataURL();
+  ref.child(currentUser).set(data)
 };
 
-var serializeOut = function() {
-  var data = ctx.toDataURL();
-  localStorage.setItem('OurCanvas', data);
+function removeCanvasAll(){
+ $('canvas').remove();
```

Comment:
This risks removing legitimate canvases from the underlying site. Can we just call removePublicCanvasAll and toggleUserCanvasOff instead?